File: internal/moleman/doctor_test.go
package moleman

import (
	"os"
	"path/filepath"
	"strings"
	"testing"
)

func TestDoctorMissingConfig(t *testing.T) {
	missingPath := filepath.Join(t.TempDir(), "missing.yaml")
	if err := Doctor(missingPath); err == nil || !strings.Contains(err.Error(), "config not found") {
		t.Fatalf("expected missing config error, got %v", err)
	}
}

func TestDoctorAnnotatesPipelineErrors(t *testing.T) {
	dir := t.TempDir()
	configPath := filepath.Join(dir, "moleman.yaml")
	config := `version: 1
agents: {}
workflow: []
`
	if err := os.WriteFile(configPath, []byte(config), 0o644); err != nil {
		t.Fatalf("write config: %v", err)
	}

	err := Doctor(configPath)
	if err == nil {
		t.Fatalf("expected validation error")
	}
	if !strings.Contains(err.Error(), "agents map is empty") {
		t.Fatalf("expected agents validation error, got %v", err)
	}
}

File: test/doctor.test.ts
import test from "node:test";
import assert from "node:assert/strict";
import { promises as fs } from "node:fs";
import os from "node:os";
import path from "node:path";
import { doctor } from "../src/doctor.js";

async function tempDir(prefix: string): Promise<string> {
  return fs.mkdtemp(path.join(os.tmpdir(), prefix));
}

test("doctor reports missing config", async () => {
  const dir = await tempDir("moleman-doctor-");
  const missingPath = path.join(dir, "missing.yaml");

  await assert.rejects(() => doctor(missingPath), /config not found/);
});

test("doctor surfaces validation errors", async () => {
  const dir = await tempDir("moleman-doctor-");
  const configPath = path.join(dir, "moleman.yaml");
  const agentsPath = path.join(dir, "agents.yaml");

  await fs.writeFile(agentsPath, "agents: {}\n", "utf8");
  await fs.writeFile(
    configPath,
    `version: 1
agents: {}
workflow: []
`,
    "utf8",
  );

  await assert.rejects(() => doctor(configPath), /agents map is empty/);
});
